### DISCUSSION TITLE
Could podman be installed and run in android?

### DISCUSSION THREAD

**Discussion by radiorz** [containers/podman#13774] [Q&A]
Could podman be installed and run in android?
I hava a useless phone，Could i use podman and make it a server？

**Comment by radiorz** [comment]
or could ubuntu "touch" install podman?

**Reply by stanekondrej** [reply]
Ubuntu Touch is definitely capable of running Podman. It's a full-blown Ubuntu installation with a mobile UI glued onto it, after all.

**Comment by rhatdan** [comment]
I have no idea.  I have never heard of anyone running Podman on Android.  Don't know if Android supports all of the namespaces required by Podman.  I don't now touch, but I would figure it would be more likely to work on a flavor of ubuntu.

**Reply by mheon** [reply]
There was an issue several years ago with someone who'd managed to get Podman running on Android. I'll try and locate it.

**Reply by Narvey** [reply]
@mheon did you find it?

**Reply by mheon** [reply]
No, sorry

**Reply by hoshsadiq** [reply]
This is an old thread, but after a little bit of research, it seems the Android kernel needs some specific things enabled to be able to run containers natively. There are some tutorials on doing this on a rooted device by recompiling the Android kernel, but that's generally quite involved and means losing some functionality (e.g. SafetyNet and Knox will fail, which will break many things).

**Comment by hoshsadiq** [comment]
I'm wondering if it would be feasible to have Podman capabilities via Podman machine. It is possible to run qemu on Android via Termux, and Google has recently enabled more KVM options in the kernel. If we run a base aarch64 fedora image on qemu similar to how Podman runs in Mac and Windows, it could be feasible.

**Reply by hoshsadiq** [reply]
Alternatively, some more information on Android's KVM implementation can be located in the docs under [Android Virtualization Framework (AVF) overview](https://source.android.com/docs/core/virtualization).

**Reply by rhatdan** [reply]
Yes I think using podman-remote on android talking to Podman running inside of a VM would work.

**Reply by jasalt** [reply]
There's a Docker installation guide with qemu VM https://github.com/cyberkernelofficial/docker-in-termux, maybe could work with minor adjustment with Podman?

**Reply by hoshsadiq** [reply]
@rhatdan do you or any of your colleagues have some pointers on how to implement this. I assume it's not high on your priority list, so I'd like to at least give it a try

**Reply by rhatdan** [reply]
I have never written any code for Android, so I would not be much help.  @jasalt above suggested you read that linked document.

**Reply by hoshsadiq** [reply]
I don't think there's any specific android related code to write. It's just a matter of updating podman to build with android as a target, and ensuring it is able to reference the VM related commands. I don't know how podman machine etc works currently for macos/windows, but it'd be essentially similar implementation.

**Comment by jasalt** [comment]
There's a guide on installing Docker directly on (rooted) Android, maybe it could give some ideas https://gist.github.com/FreddieOliveira/efe850df7ff3951cb62d74bd770dce27.
